windows-agent/internal/proservices/landscape/executor.go

```diff
@@ -189,3 +196,119 @@ func (e executor) setDefault(ctx context.Context, cmd *landscapeapi.Command_SetD
 func (e executor) shutdownHost(ctx context.Context, cmd *landscapeapi.Command_ShutdownHost) error {
 	return gowsl.Shutdown(ctx)
 }
+
+func installFromMicrosoftStore(ctx context.Context, distro gowsl.Distro) error {
+	if err := gowsl.Install(ctx, distro.Name()); err != nil {
+		return err
+	}
+
+	if err := distroinstall.InstallFromExecutable(ctx, distro); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func installFromURL(ctx context.Context, homeDir string, distro gowsl.Distro, rootfs *landscapeapi.Command_Install_Rootfs) (err error) {
+	// Since we are going to remove the tarball soon, I'd say it's fine to write it to a subfolder of the current working directory,
+	// as it will be somewhere inside the Windows Apps private folders:
+	// %USERPROFILE%\AppData\Local\Packages\CanonicalGroupLimited.UbuntuPro_79rhkp1fndgsc\LocalState\
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	tmpDir := filepath.Join(cwd, distro.Name())
+	if err := os.MkdirAll(tmpDir, 0700); err != nil {
+		return err
```

Comment:
I'm not sure what you mean here, but take a look at my recent change and see if that's ok.

Reply:
There is a test case in which we break the creation of the tarball by creating a directory with the same name as the file that would be created, causing this function to fail, bailing out in line 229.
Didier suggested us to add another test case in which, instead of breaking `os.Create(tarball)` we'd break `os.MkDirAll(tmpDir,...` by creating a file with the same name, thus causing the line 220 to be covered.

Reply:
Oh ok, I was misreading the original message. Most recent commit should add a test for that

Reply:
yeah, that was it!